kube-client/src/config/file_loader.rs

```diff
@@ -91,8 +91,8 @@ impl ConfigLoader {
 
         Ok(ConfigLoader {
             current_context: current_context.clone(),
-            cluster: cluster.clone(),
-            user: user.clone(),
+            cluster: cluster.clone().unwrap_or_default(),
+            user: user.clone().unwrap_or_default(),
```

Comment:
I think we should do the following instead of `unwrap_or_default()`:

```rust
        let cluster = config
            .clusters
            .iter()
            .find(|named_cluster| &named_cluster.name == cluster_name)
            // change `map` to `and_then` and clone
            .and_then(|named_cluster| named_cluster.cluster.clone())
            .ok_or_else(|| KubeconfigError::LoadClusterOfContext(cluster_name.clone()))?;

        let user_name = user.unwrap_or(&current_context.user);
        let user = config
            .auth_infos
            .iter()
            .find(|named_user| &named_user.name == user_name)
            // change `map` to `and_then` and clone
            .and_then(|named_user| named_user.auth_info.clone())
            .ok_or_else(|| KubeconfigError::FindUser(user_name.clone()))?;

        Ok(ConfigLoader {
            current_context: current_context.clone(),
            // remove clone
            cluster,
            user,
        })
```

Reply:
Indeed, that's a lot cleaner, thanks your suggestion! I've pushed the change.